doc/nrf/app_dev/tfm/index.rst

```diff
@@ -149,6 +151,49 @@ Nordic platforms support only the GCC toolchain for building TF-M.
 Enabling Floating point support in TF-M is currently not supported.
 Enabling Floating point Hard ABI (:kconfig:option:`CONFIG_FP_HARDABI`) in the application is currently not supported.
 
+.. _ug_tfm_partition_alignment_requirements:
+
+TF-M partition alignment requirements
+*************************************
+
+TF-M requires that secure and non-secure partition addresses must be aligned to the NRF_SPU flash region size :kconfig:option:`CONFIG_NRF_SPU_FLASH_REGION_SIZE`.
+|NCS| ensures that they in fact are aligned and comply with the TF-M requirements.
+
+TF-M requires this alignment because it uses the SPU to enforce the security policy between the partitions.
+When the partition manager is enabled, it will take into consideration the alignment requirements.
+But when the static partitions are used, the user is responsible for following the alignment requirements.
+
+If you are experiencing any partition alignment issues when using the partition manager, check for known issues.
+
+The partitions which need to be aligned to the SPU flash region size are the tfm_secure and tfm_nonsecure partitions.
+
+The following static partition snippet shows a non-aligned configuration for nRF5340 which has a SPU flash region size :kconfig:option:`CONFIG_NRF_SPU_FLASH_REGION_SIZE` of 0x4000.
```

Comment:
The example is for static partitions, we should have one where a static partition is not used.

When using a static partition file:

<example>

When using a generated partition file:

Change value of CONFIG_PM_PARTITION_SIZE_TFM (or other PM_PARTITION_SIZE) configurations so that the end of TF-M is aligned on ...
Compare the generated partition file <build>/partitions.yml with the above example for static partitions.

Reply:
I think that this will complicate things more and it is not meant to exist in the application development. It is the responsibility of the partition manager to make sure that this works as expected, not a requirement of the user. In the known issues with your addition we mention this configuration change so for me this is enough.
Let me know if you think that this is required.